src/dependency_injector/providers.pyi

```diff
@@ -249,9 +250,9 @@ class Factory(Provider[T]):
     provided_type: Optional[Type]
     def __init__(self, provides: Optional[_Callable[..., T]] = None, *args: Injection, **kwargs: Injection) -> None: ...
     @property
-    def cls(self) -> T: ...
+    def cls(self) -> Type[T]: ...
     @property
-    def provides(self) -> T: ...
+    def provides(self) -> Type[T]: ...
```

Comment:
Like in the case with ``Callable`` provider we better return ``_Callable[..., T]`` here. With the current design ``Factory`` and ``Singleton`` providers can take not only classes but any callables. This change design change was made to support factory method and functions like ``sqlite.connect()``, ``logging.getLogger()``, etc.

Reply:
done 47cf4c2